audit-service/src/unit-tests/java/uk/gov/gchq/palisade/service/audit/AbstractSerialisationTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.audit;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.skyscreamer.jsonassert.JSONAssert.assertEquals;
+
+/**
+ * Abstract class providing a serialisation test for a class
+ */
+@SuppressWarnings("java:S2187") // no tests in this class by design
+public class AbstractSerialisationTest {
+
+    private static ObjectMapper mapper;
+
+    @BeforeAll
+    static void setupAll() {
+        mapper = new ObjectMapper().registerModules(new Jdk8Module());
+    }
+
+    @AfterAll
+    static void afterAll() {
+        mapper = null;
+    }
+
+    /**
+     * Tests the serialisation and deserialisation of each object provided by the
+     * method source
+     * <p>
+     * The tests asserted by this method:
+     * <ul>
+     * <li><em>Equals</em> - Test equality using the each instances equals
+     * method</li>
+     * <li><em>Recursive Equals</em> - Tests equality by testing field by field in
+     * the instance</li>
+     * </ul>
+     *
+     * @param type             The type class
+     * @param <O>              The type of the object itself
+     * @param <T>              The type of the object
+     * @param expectedInstance The expected instance
+     * @return The object being tested
+     * @throws Exception if an error occurs
+     */
+    protected <O, T extends O> T testInstance(final Class<T> type, final O expectedInstance) throws Exception {
+        return testInstance(type, expectedInstance, null);
```

Comment:
why is there a null here? just remove if its always unused

Reply:
Need the null as the method is type, instance and JSON string. The called method tests to see if it should actually perform the JSON equals with JSONAssert.